ocw/lib/azure.py

```diff
@@ -157,25 +163,41 @@ def cleanup_blob_containers(self) -> None:
                             self.log_info(f"Deleting blob {blob.name}")
                             self.container_client(container.name).delete_blob(blob.name, delete_snapshots="include")
 
-    def cleanup_images_from_rg(self) -> None:
-        self.log_dbg("Call cleanup_images_from_rg")
-        for item in self.list_images_by_resource_group(self.__resource_group):
+    def cleanup_images(self) -> None:
+        self.log_dbg("Call cleanup_images")
+        for item in self.list_images():
             if self.is_outdated(item.changed_time):
                 if self.dry_run:
                     self.log_info(f"Deletion of image {item.name} skipped due to dry run mode")
                 else:
                     self.log_info(f"Delete image '{item.name}'")
                     self.compute_mgmt_client().images.begin_delete(self.__resource_group, item.name)
 
-    def cleanup_disks_from_rg(self) -> None:
-        self.log_dbg("Call cleanup_disks_from_rg")
-        for item in self.list_disks_by_resource_group(self.__resource_group):
+    def cleanup_disks(self) -> None:
+        self.log_dbg("Call cleanup_disks")
+        for item in self.list_disks():
             if self.is_outdated(item.changed_time):
                 if self.compute_mgmt_client().disks.get(self.__resource_group, item.name).managed_by:
-                    self.log_warn(f"Disk is in use - unable delete {item.name}")
+                    self.log_warn(f"Disk is in use - skipping {item.name}")
                 else:
                     if self.dry_run:
                         self.log_info(f"Deletion of disk {item.name} skipped due to dry run mode")
                     else:
                         self.log_info(f"Delete disk '{item.name}'")
                         self.compute_mgmt_client().disks.begin_delete(self.__resource_group, item.name)
+
+    def cleanup_versions(self) -> None:
+        self.log_dbg("Call cleanup_versions")
+        for gallery in self.list_galleries():
```

Comment:
I would not do the version cleanup in all galleries. We only have one specified Image Gallery for our tests.
I would configure the image gallery in the configuration file, and perform the cleanup only on this one, so that we can have test galleries, which are not being affected. This was useful while developing and might come in handy, when we have to do some testing.

In addition I would introduce a check, that if the `pcw_ignore` tag is applied to a image gallery, then the whole gallery and it's definitions will be skipped.

Reply:
> I would not do the version cleanup in all galleries. We only have one specified Image Gallery for our tests. I would configure the image gallery in the configuration file, and perform the cleanup only on this one, so that we can have test galleries, which are not being affected. This was useful while developing and might come in handy, when we have to do some testing.

Fixed.

> In addition I would introduce a check, that if the `pcw_ignore` tag is applied to a image gallery, then the whole gallery and it's definitions will be skipped.

I went further and introduced tag checking also for images and versions.